Source/Core/InputCommon/ControllerInterface/CoreDevice.h

```diff
@@ -118,7 +118,8 @@ class Device
   virtual std::string GetName() const = 0;
   virtual std::string GetSource() const = 0;
   std::string GetQualifiedName() const;
-  virtual void UpdateInput() {}
+  // Return false to remove the device
+  virtual bool UpdateInput() { return true; }
```

Comment:
I think this `bool` should be replaced with e.g. `enum class DeviceRemoval { RemoveDevice, KeepDevice };`. This would make the purpose of the return value much more clear.